phpseclib/Crypt/Random.php

```diff
@@ -148,13 +148,13 @@ function crypt_random_string($length)
             session_start();
 
             $v = $seed = $_SESSION['seed'] = pack('H*', sha1(
-                serialize($_SERVER) .
-                serialize($_POST) .
-                serialize($_GET) .
-                serialize($_COOKIE) .
-                serialize($GLOBALS) .
-                serialize($_SESSION) .
-                serialize($_OLD_SESSION)
+                phpseclib_safe_serialize($_SERVER) .
+                phpseclib_safe_serialize($_POST) .
+                phpseclib_safe_serialize($_GET) .
+                phpseclib_safe_serialize($_COOKIE) .
+                phpseclib_safe_serialize($GLOBALS) .
+                phpseclib_safe_serialize($_SESSION) .
+                phpseclib_safe_serialize($_OLD_SESSION)
```

Comment:
Crypt/Random.php does `session_start()` before trying to call `$_SESSION`.  Normally that'd create `$_SESSION` but not on http://sandbox.onlinephpfunctions.com/ and that's because of the php.ini thing you mentioned. eg.
```php
<?php
session_start();
serialize($_SESSION);
```

That yields a `Warning: session_start() has been disabled for security reasons` error:

http://sandbox.onlinephpfunctions.com/code/15254dd8e75ef36649a2afae66470662d8782c64

So they probably have that function mentioned in `disable_functions`:

http://php.net/manual/en/ini.core.php#ini.disable-functions

I guess we could still do `isset()` but idk...  issues arising from disabled functions are not issues that we should be overly concerned with imho. You disable functions at your own peril.